### DISCUSSION TITLE
Issues with round objects

### DISCUSSION THREAD

**Discussion by Gurbalur** [feelfreelinux/octo4a#257] [Q&A]
MiniX Neo X7. I finally got it to work but when printing round objets, curves etc the printer judders and jerks instead of printing a smooth round/curved line. Works fine when printing with SD in the printer. Was recommended changing Max resolution in Cura but that did nothing. Tried different USB cable but then the printer wasn't recognized. I don't know how much of a speed the cable must support but these mini USBs are getting old. I have printed with the same cable via Cura and had no such issues. Any suggestions?

**Comment by mattcasper78** [comment]
i just googled and found this- i'm having the same issue printing any type of cylinder. I've been through every setting for days. 
Even a vase stutters and leaves a blob at random intervals.

**Reply by KStag13** [reply]
I used to have a similar issue when I was using a cheap cable between a computer and the printer. Basically the printer runs out of GCode to dun before more is sent over. The solution then was of course to switch to using the sd card. Maybe a high quality cord would help. Though I don't know how good you can get with a micro USB.

**Reply by Gurbalur** [reply]
Maybe. I just used the cable that came with the printer.

**Comment by Jonatassfredo** [comment]
> MiniX Neo X7. I finally got it to work but when printing round objets, curves etc the printer judders and jerks instead of printing a smooth round/curved line. Works fine when printing with SD in the printer. Was recommended changing Max resolution in Cura but that did nothing. Tried different USB cable but then the printer wasn't recognized. I don't know how much of a speed the cable must support but these mini USBs are getting old. I have printed with the same cable via Cura and had no such issues. Any suggestions?

I've same problem on Round Objects. It's a Octoprint Bug?

**Comment by Thomahawkuru** [comment]
*EDIT: Using the Arc-Welder plugin largely solved this problem for me. See reactions on this reply and [issue #123](https://github.com/feelfreelinux/octo4a/issues/123) for more details* 

Yup, i have the same problem, stuttering around curves leaving blobs. I have tried a couple things with little but some succes:

Increased serial buffer size in marlin firmware, thinking it ran out. no change
decreased functionality in firmware to increase free available dynamic memory. no change
disabled arc suport in marlin. no change
ran octoprint on my daily phone, instead of my previous older model. Problem goes away!
So, to me it looks like its related to the capabilities of the phone maybe. I ran all phones with power saving disabled:

I first ran a galaxy a50 when the problem occured.
Then tried a galaxy s7 edge, with the same issue but it seemed to occure less.
Then I tried my daily galaxy s21 FE and the issue went away
I seems to me that all those phones should have the hardware capable of running octoprint, but the issue seems to be related to how powerfull the phone is?

What phones are you running this on?

**Reply by Gurbalur** [reply]
Could be. I don't use a phone but an Android media box.
https://www.minix.us/products/neox7.html

**Reply by Thomahawkuru** [reply]
Ah I see, you declared it in the original post but I must have skipped over it. Still I think its device/octoprint performance related

I did some further digging and testing and I found a solution in the issues tab here on the GitHub ([issue #123](https://github.com/feelfreelinux/octo4a/issues/123)).  The solution is to use the plugin Arc Welder, available for octoprint and cura (and maybe more but i'm not aware). This plugin will process the gcode and combine all the small line segments of a curve into one single curved line. However your firmware must support such commands with G2 and G3 (arc_support in marlin). 

I was running marlin 2.0.5.3 at first, which does have arc_support, however it made literally all curves extremely jittery. I read somewhere that the implementation of arc_support isn't great on earlier marlin versions, so I upgraded to 2.0.9.5 and now it works great. It's still not perfect, because at very small features with a lot of tiny movements I still get the occasional hiccup, but all the curves are smooth!

Maybe give it a try and see if it works for you. Good Luck!

*EDIT: using the new firmware 2.0.9.5 without arc-welder did not improve the issue for me*

**Reply by Gurbalur** [reply]
You may be right. I haven had much luck with arc welder previously as I got a lot of artifacts but I will try it. Thanks.
